apps/web-portal-client/README.md

```diff
@@ -0,0 +1,128 @@
+# web-portal-client
+
+
+## Start Project in Dev Mode
+
+`yarn start`
+
+
+## Design Decisions
+
+We choose to build our client on top of React + [Redux Tooltip](https://redux-toolkit.js.org/) stack
+
+Application consist within  `Features` each feature may has next layers
+
+1. `Slices` -  provides reducer and actions to manipulate within particular feature state slice
+1. `Selectors` - provides access to data required by a feature related to it's state slice
+1. `Effects` - contains interactions with side effects
+1. `Middlewares` - Redux middleware, main propose is to listen for app action flow and to some additional actions
+1. `Components` - 'dummy' components, which has no relations for `Slices` and `Selectors`
+1. `Containers` - components used by external features, usually use `Selectors` and Redux Actions
+
+### Features
+
+Parts of application which may contains UI. Feature usually contains store state, actions to work with that store and
+already 'connected' UI component to application store.
+
+
+### Slices
+
+Read [Slice](https://redux.js.org/tutorials/fundamentals/part-8-modern-redux#using-createslice)
+
+Each slice is a module which combine action creators and single reducer for particular application store slice
+
+Naming of files inside of slice folder
+
+```
+|-slices/
+  |-index.js                   // exports all actions and 'reducer' for current slice
+  |-mySliceNameReducer.js      // main reducer for current slice
+  |-mySliceActions.js          // all actions creators (no any think actions here)
+```
+
+Example: usage of slice
+```
+import {reducer, specsRemoveSuccess } from './slices/specs'
+
+```
+### Effects
+
+Effects in that projects are [redux-thunk](https://github.com/reduxjs/redux-thunk) functions which trigger side effects.
+Like API calls, timers etc.  In case of middleware, put it here too
+
+for example:
+
+```
+|-effects/
+  |-index.js                   // exports all actions and 'reducer' for current slice
+  |-openSpecEffect.js          // main reducer for current slice
+  |-editSpecEffect.js          // all actions creators (no any think actions here)
+  |-someMiddleware.js          // some middleware
+```
+
+### Selectors
+
+Selector functions for read-only operation over root app state.
+Should use cache to optimize React re-rendering.
+
+### Components
+
+*Reusable* React components, shouldn't have any dependencies to Redux, try to keep them as simple as possible
```

Comment:
The component that has a connection to Redux is a container, maybe somehow emphasize this? The statement little bit confusing. Btw, are we allow HOCs to be connected to the Redux?

Reply:
Should we add some rules for avoiding the heavy calculations in the render method,
Redux usage for avoiding extra rerenders, and other rules,
or this is not a proper place for this?

Reply:
> Btw, are we allow HOCs to be connected to the Redux?

Only in some exceptional situations. In most cases, we are going to avoid HOCs in general and HOCs to Redux 

> The component that has a connection to Redux is a container, maybe somehow emphasize this?
yep, it's in 'Container' section